openedx-unsupported/configuration
# SE-2298 Makes the ecommerce OSCAR_DEFAULT_CURRENCY configurable.

(cherry picked from commit 436092e2ab1a4dd8db3e5e077dfb6d8dd7151d71)

This allows overriding the default currency in ecommerce during deployment. Useful for instances where the desired currency isn't USD. Originally introduced in https://github.com/open-craft/configuration/pull/119

**Jira tickets**: [OSPR-4409](https://openedx.atlassian.net/browse/OSPR-4409)

**Dependencies**: For this to successfully configure ecommerce, ecommerce requires the changes in https://github.com/edx/ecommerce/pull/2916

**Sandbox URL**: 

-    LMS: https://ecommercepr2916.sandbox.opencraft.hosting/
-    Studio: https://studio.ecommercepr2916.sandbox.opencraft.hosting/

**Merge deadline**: None

**Test instructions**:

1. Set `ECOMMERCE_OSCAR_DEFAULT_CURRENCY` and deploy a new appserver using this branch and the ecommerce branch from https://github.com/edx/ecommerce/pull/2916.
1. Ensure that the ecommerce `settings.OSCAR_DEFAULT_CURRENCY` matches the setting.

**Reviewers**:


**Settings**:
```yaml
SANDBOX_ENABLE_ECOMMERCE: yes
ECOMMERCE_OSCAR_DEFAULT_CURRENCY: "GBP"
ECOMMERCE_VERSION: samuel/override-default-currency
ECOMMERCE_REPOS:
  - PROTOCOL: "{{ COMMON_GIT_PROTOCOL }}"
    DOMAIN: "{{ COMMON_GIT_MIRROR }}"
    PATH: "open-craft"
    REPO: "ecommerce.git"
    VERSION: "{{ ECOMMERCE_VERSION }}"
    DESTINATION: "{{ ecommerce_code_dir }}"
    SSH_KEY: "{{ ECOMMERCE_GIT_IDENTITY }}"
```
